stdlib/Random/test/runtests.jl

```diff
@@ -801,3 +801,29 @@ end
 @testset "RNGs broadcast as scalars: T" for T in (MersenneTwister, RandomDevice)
     @test length.(rand.(T(), 1:3)) == 1:3
 end
+
+const BASE_TEST_PATH = joinpath(Sys.BINDIR, "..", "share", "julia", "test")
+isdefined(Main, :OffsetArrays) || @eval Main include(joinpath($(BASE_TEST_PATH), "testhelpers", "OffsetArrays.jl"))
+
+@testset "OffsetArrays" begin
+    # axes as tuple
+    @test axes(rand(Bool, (Base.OneTo(3), Base.OneTo(3)))) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, (Base.OneTo(3), 3))) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, (0:2, 0:2))) == (0:2, 0:2)
+
+    # 1D axes
+    @test axes(rand(Bool, Base.OneTo(3))) === (Base.OneTo(3), )
+    @test axes(rand(Bool, 3)) === (Base.OneTo(3), )
+    @test axes(rand(Bool, 0:2)) == (0:2, )
+
+    # axes splatted
+    @test axes(rand(Bool, Base.OneTo(3), Base.OneTo(3))) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, Base.OneTo(3), 3)) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, 0:2, 0:2)) == (0:2, 0:2)
+
+    # edge cases
+    @test rand(3) isa Array{Float64, 1}
+    @test rand(1:3) isa Integer
+    @test rand(1:3, 1:3) isa OffsetArrays.OffsetArray
```

Comment:
It looks good to me but I'm not sure how others look at this. This behavior seems not very compatible with the current usage of `rand`; it is reasonable but it would be a little unintuitive that `rand(3, 1:3)` returns an offset matrix.

Reply:
To check that I understand, what would be more intuitive for `rand(3, 1:3)`?

Reply:
I was thinking the same you mentioned in #36557, this should be the correct behavior considering the backward compatibility.

Reply:
The example there was more like `rand(1:3)` than `rand(3, 1:3)`